packages/apollo-link-utilities/src/index.ts

```diff
@@ -0,0 +1,247 @@
+import { Operation } from 'apollo-link';
+import { print } from 'graphql/language/printer';
+
+// XXX replace with actual typings when available
+declare var AbortController: any;
+
+//Used for any Error for data from the server
+//on a request with a Status >= 300
+//response contains no data or errors
+export type ServerError = Error & {
+  response: Response;
+  result: Record<string, any>;
+  statusCode: number;
+};
+
+//Thrown when server's resonse is cannot be parsed
+export type ServerParseError = Error & {
+  response: Response;
+  statusCode: number;
+  bodyText: string;
+};
+
+export type ClientParseError = Error & {
+  parseError: Error;
+};
+
+export interface HttpOptions {
+  includeQuery?: boolean;
+  includeExtensions?: boolean;
+}
+
+export interface ConfigOptions {
+  http?: HttpOptions;
+  options?: any;
+  headers?: any; //overrides headers in options
+  credentials?: any;
+}
+
+export namespace LinkUtils {
```

Comment:
Would it not be better to export everything in this `LinkUtils` namespace separately? At the moment we can't use a named import to just grab what we want (`fallbackConfig` is all I need for `apollo-upload-client`), defeating tree-shaking. `LinkUtils` is also not a very intuitive named import, looking at it you might think it is some sort of class that can be constructed, or that it contains everything in `apollo-link-utilities`. `Utils` sounds like helper functions, but the namespace contains a lot of default options objects.

Reply:
That makes sense. I wasn't aware of the lack of tree-shaking. Thanks!

You also bring up a good point about the naming. I think there is a general issue in the utilities, so I am going to make the utility functions a bit more descriptive.